plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,697 @@
+    storage: local-zfs
+    size: 5
+    state: present
+
+- name: Create new disk in VM (force rewrite in case it exists already)
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: scsi3
+    format: qcow2
+    storage: local
+    size: 16
+    force_replace: True
+    state: present
+
+- name: Update existing disk
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: ide0
+    backup: False
+    ro: True
+    aio: native
+    state: updated
+
+- name: Grow existing disk
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: sata4
+    size: +5G
+    state: grown
+
+- name: Detach disk (leave it unused)
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    name: vm-name
+    disk: virtio0
+    state: detached
+
+- name: Move disk to another storage
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_password: secret
+    vmid: 101
+    disk: scsi7
+    target_storage: local
+    format: qcow2
+    state: moved
+
+- name: Move disk from one VM to another
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_token_id: token1
+    api_token_secret: some-token-data
+    vmid: 101
+    disk: scsi7
+    target_vmid: 201
+    state: moved
+
+- name: Remove disk permanently
+  community.general.proxmox_disk:
+    api_host: node1
+    api_user: root@pam
+    api_password: secret
+    vmid: 101
+    disk: scsi4
+    state: absent
+'''
+
+RETURN = '''
+vmid:
+  description: The VM vmid.
+  returned: success
+  type: int
+  sample: 101
+msg:
+  description: A short message
+  returned: always
+  type: str
+  sample: "Disk scsi3 created in VM 101"
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible_collections.community.general.plugins.module_utils.proxmox import (proxmox_auth_argument_spec,
+                                                                                ProxmoxAnsible)
+from re import compile, match, sub
+from time import sleep
+
+
+def disk_conf_str_to_dict(config_string):
+    config = config_string.split(',')
+    storage_volume = config.pop(0).split(':')
+    config.sort()
+    storage_name = storage_volume[0]
+    volume_name = storage_volume[1]
+    config_current = dict(
+        volume='%s:%s' % (storage_name, volume_name),
+        storage_name=storage_name,
+        volume_name=volume_name
+    )
+
+    for i in config:
+        kv = i.split('=')
+        try:
+            config_current[kv[0]] = kv[1]
+        except IndexError:
+            config_current[kv[0]] = ''
```

Comment:
I ran a test locally:
```
>>> k, v = "a=3".split("=")
>>> k, v
('a', '3')
>>> k, v = "a=".split("=")
>>> k, v
('a', '')
```
So this could be simplified to (and changing the name of the loop variable):
```suggestion
    for cfg in config:
        k, v = cfg.split('=')
        config_current[k] = v
```